addons/sourcemod/scripting/shavit-core.sp

```diff
@@ -2159,11 +2064,169 @@ public Action OnPlayerRunCmd(int client, int &buttons, int &impulse, float vel[3
 			}
 		}
 	}
+}
 
-	gI_ButtonCache[client] = buttons;
-	gF_AngleCache[client] = angles[1];
+void CountStrafes(int client, int buttons)
+{
+	if(gA_StyleSettings[gBS_Style[client]][bStrafeCountW] && !gA_StyleSettings[gBS_Style[client]][bBlockW] &&
+		(gI_ButtonCache[client] & IN_FORWARD) == 0 && (buttons & IN_FORWARD) > 0)
+	{
+		gI_Strafes[client]++;
+	}
 
-	return Plugin_Continue;
+	if(gA_StyleSettings[gBS_Style[client]][bStrafeCountA] && !gA_StyleSettings[gBS_Style[client]][bBlockA] && (gI_ButtonCache[client] & IN_MOVELEFT) == 0 &&
+		(buttons & IN_MOVELEFT) > 0 && (gA_StyleSettings[gBS_Style[client]][iForceHSW] > 0 || ((buttons & IN_FORWARD) == 0 && (buttons & IN_BACK) == 0)))
+	{
+		gI_Strafes[client]++;
+	}
+
+	if(gA_StyleSettings[gBS_Style[client]][bStrafeCountS] && !gA_StyleSettings[gBS_Style[client]][bBlockS] &&
+		(gI_ButtonCache[client] & IN_BACK) == 0 && (buttons & IN_BACK) > 0)
+	{
+		gI_Strafes[client]++;
+	}
+
+	if(gA_StyleSettings[gBS_Style[client]][bStrafeCountD] && !gA_StyleSettings[gBS_Style[client]][bBlockD] && (gI_ButtonCache[client] & IN_MOVERIGHT) == 0 &&
+		(buttons & IN_MOVERIGHT) > 0 && (gA_StyleSettings[gBS_Style[client]][iForceHSW] > 0 || ((buttons & IN_FORWARD) == 0 && (buttons & IN_BACK) == 0)))
+	{
+		gI_Strafes[client]++;
+	}
+}
+
+#if defined DEBUG
+void Debug_CheckVelocity(int client, float vel[3])
+{
+	static int cycle = 0;
+
+	if(++cycle % 50 == 0)
+	{
+		Shavit_StopChatSound();
+		Shavit_PrintToChat(client, "vel[0]: %.01f | vel[1]: %.01f", vel[0], vel[1]);
+	}
+}
+#endif
+
+void BlockButtons(int client, int buttons, float vel[3], int iGroundEntity)
```

Comment:
`int buttons` >> `int &buttons`

`iGroundEntity` >> `groundentity`, as it's a parameter. should be consistent with the rest of the code